### PR TITLE
Nom 1635 vector search in js

### PR DESCRIPTION
**Pull request by bmschmidt**
Use new endpoints.

When called from index, gives two matched lists-of-lists; when called from projection, gives a list of closest matches for each point.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -113,4 +114,29 @@ export class AtlasIndex extends BaseAtlasClass {
...
+  >): Promise<components['schemas']['EmbeddingNeighborResponse']> {
+    const { neighbors, distances } = (await this.apiCall(
+      `/v1/project/data/get/nearest_neighbors/by_embedding`,
+      'POST',
+      {
+        atlas_index_id: this.id,
+        k,
+        queries,
+      }
+    )) as components['schemas']['EmbeddingNeighborResponse'];
```
Comment (socalledso) [inline src/index.ts:139]: Instead of doing an 'as' and force type checking. Let's use 'satisfies'

File: src/project.ts [hunk 2]
```diff
@@ -220,8 +220,19 @@ export class AtlasDataset extends BaseAtlasClass {
...
+    if (ids === undefined) {
+      return {};
+    }
+    const response = await this.apiCall(
+      '/v1/project/data/get',
+      'POST',
+      { project_id: this.id, datum_ids: ids },
+      null
+    );
+    return response as Record<string, Record<string, any>>;
```
Comment (socalledso) [inline src/project.ts:235]: Do we need this as? I figured the Promise wrapper itself puts a type. The second assertion doesn't do anything

File: src/projection.ts [hunk 3]
```diff
@@ -300,6 +301,42 @@ export class AtlasProjection extends BaseAtlasClass {
...
+    for (let i = 0; i < neighbors.length; i++) {
+      filled_out[i] = [];
+      for (let j = 0; j < neighbors[i].length; j++) {
+        const d = { ...datums[i][j] };
+        d._distance = distances[i][j];
+        filled_out[i].push(d);
+      }
+    }
+
+    console.log({ filled_out });
```
Comment (socalledso) [inline src/projection.ts:335]: Stale console log.

There are also two returns?

**Review by socalledso — CHANGES_REQUESTED** [review]
Left few comments

File: src/index.ts (hunk 1 above)
Comment (bmschmidt) [reply src/index.ts:139]: If you can show a way that this works please do, I don't know what you mean here.

File: src/project.ts (hunk 2 above)
Comment (bmschmidt) [reply src/project.ts:235]: We do need this to pass type checking, yes.

File: src/projection.ts (hunk 3 above)
Comment (bmschmidt) [reply src/projection.ts:335]: Fixed

File: src/index.ts (hunk 1 above)
Comment (bmschmidt) [reply src/index.ts:139]: The issue is that API call can return many different forms, so satisfies fails type checking.

**Review by socalledso — APPROVED** [review]

**Review by AndriyMulyar — APPROVED** [review]

**Pull request merged**